libs/platform/src/lib/components/form/input/input.component.ts

```diff
@@ -34,12 +50,39 @@ export type InputType = 'text' | 'number' | 'email';
     selector: 'fdp-input',
     templateUrl: 'input.component.html',
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [{ provide: FormFieldControl, useExisting: InputComponent, multi: true }]
+    providers: [
+        { provide: FormFieldControl, useExisting: InputComponent, multi: true }
+    ]
 })
 export class InputComponent extends BaseInput {
+
+    /** defines the input type of the input. */
     @Input()
     type: InputType = 'text';
+ 
+    /** Whether the input is read-only. */
+    @Input()
+    readonly: boolean = false;
```

Comment:
I think we should propagate readonly to parent class as there will be more use of it